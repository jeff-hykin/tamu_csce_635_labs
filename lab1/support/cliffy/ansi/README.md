# Ansi

<p align="center" class="badges-container">
  <a href="https://github.com/c4spar/deno-cliffy/releases">
    <img alt="Version" src="https://img.shields.io/github/v/release/c4spar/deno-cliffy?logo=github&color=blue" />
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/actions/workflows/test.yml">
    <img alt="Build status" src="https://github.com/c4spar/deno-cliffy/workflows/Test/badge.svg?branch=main" />
  </a>
  <a href="https://codecov.io/gh/c4spar/deno-cliffy">
    <img src="https://codecov.io/gh/c4spar/deno-cliffy/branch/main/graph/badge.svg"/>
  </a>
  <a href="https://github.com/c4spar/deno-cliffy/labels/module%3Aansi-escape">
    <img alt="issues" src="https://img.shields.io/github/issues/c4spar/deno-cliffy/module:ansi-escape?label=issues&logo=github&color=yellow">
  </a>
  <a href="https://deno.land/">
    <img alt="Deno version" src="https://img.shields.io/badge/deno-^1.4.0-blue?logo=deno" />
  </a>
  <a href="https://doc.deno.land/https/deno.land/x/cliffy/ansi/mod.ts">
    <img alt="doc" src="https://img.shields.io/badge/deno-doc-yellow?logo=deno" />
  </a>
  <a href="https://discord.gg/ghFYyP53jb">
    <img alt="Discord" src="https://img.shields.io/badge/join-chat-blue?logo=discord&logoColor=white" />
  </a>
  <a href="../LICENSE">
    <img alt="License" src="https://img.shields.io/github/license/c4spar/deno-cliffy?logo=github" />
  </a>
  <br>
  <a href="https://deno.land/x/cliffy">
    <img alt="Discord" src="https://img.shields.io/badge/Published on deno.land-blue?logo=deno&logoColor=959DA6&color=272727" />
  </a>
</p>

The documentation has been moved to https://cliffy.io/docs/ansi.
