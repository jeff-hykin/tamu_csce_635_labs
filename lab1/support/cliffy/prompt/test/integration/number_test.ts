import { ansi } from "../../../ansi/ansi.ts";
import { Number } from "../../number.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "number prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter a number": {
      stdin: ansi
        .text("19")
        .cursorUp
        .text("u")
        .cursorDown
        .cursorDown
        .cursorDown
        .cursorDown
        .cursorUp
        .cursorUp
        .cursorUp
        .text("\n")
        .toArray(),
    },
  },
  async fn() {
    await Number.prompt({
      message: "How old are you?",
      default: 7,
      max: 20,
      min: 18,
    });
  },
});

await snapshotTest({
  name: "number prompt with float",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter a floating number": {
      stdin: ansi
        .text("19.")
        .cursorUp
        .cursorUp
        .cursorUp
        .cursorUp
        .text("\n")
        .toArray(),
    },
  },
  async fn() {
    await Number.prompt({
      message: "How old are you?",
      default: 7,
      max: 20,
      min: 18,
      float: true,
    });
  },
});
