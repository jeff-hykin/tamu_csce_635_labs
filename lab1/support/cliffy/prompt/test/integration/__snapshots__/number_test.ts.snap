export const snapshot = {};

snapshot[`number prompt > should enter a number 1`] = `
stdout:
"? How old are you? (7) › \\x1b[26G\\x1b[G\\x1b[0J
? How old are you? (7) › 19\\x1b[28G\\x1b[G\\x1b[0J
? How old are you? (7) › 20\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 21\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 20\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 19\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 18\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 17\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 18\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 19\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 20\\x1b[27G\\x1b[G\\x1b[0J
? How old are you? (7) › 20
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;

snapshot[`number prompt with float > should enter a floating number 1`] = `
stdout:
"? How old are you? (7) › \\x1b[26G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.\\x1b[29G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.1\\x1b[29G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.2\\x1b[29G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.3\\x1b[29G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.4\\x1b[29G\\x1b[G\\x1b[0J
? How old are you? (7) › 19.4
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
